### ISSUE TITLE
Request: More Customization

### ISSUE THREAD

**Issue by Braunson** [surveyjs/surveyjs#949]
### Are you requesting a feature, reporting a bug or ask a question?

- Requesting a feature, allowing customization of the survey template itself, being able to add custom classes to question elements, etc.

### What is the expected behavior?

- Being able to customize the template, or in this specific case, being able to tie another existing button to be used as the 'next/complete' button.

### How would you reproduce the current behavior (if this is a bug)?

N/A

**Comment by andrewtelnov** [comment]
@Braunson, have you seen [this example](https://surveyjs.io/Examples/Library/?id=survey-customnavigation)

Thank you,
Andrew

**Comment by tsv2013** [comment]
@Braunson And these:
https://surveyjs.io/Examples/Library/?id=survey-customcss&platform=jQuery&theme=default
https://surveyjs.io/Examples/Library/?id=survey-cssclasses&platform=jQuery&theme=default

**Comment by Braunson** [comment]
@andrewtelnov Thanks but that example is for jQuery, I've tried it with Vue with no luck. Also I'm curious why it defined the title as false twice? Is that a typo?

Also why is the function `doOnCurrentPageChnged` called twice, once in the Survey call and again called by itself?

**Comment by andrewtelnov** [comment]
@Braunson Here is [the vue example](https://surveyjs.io/Examples/Library/?id=survey-customnavigation&platform=Vue)
And have you seen the [PanelDynamic Question](https://surveyjs.io/Examples/Library/?id=questiontype-paneldynamic&platform=Vue)? 
You may find more information in [this article on medium](https://hackernoon.com/surveyjs-an-open-source-javascript-survey-library-with-versions-for-angular-2-react-vue-2086d988b974), paragraph: Dynamic Panel: Dealing with List Data

Thank you,
Andrew

**Comment by Braunson** [comment]
@andrewtelnov Yes I've read through that article, thank you however I'm not looking to use Dynamic panels as there is an existing use of the 'contact' data that I don't want to import into dynamic panels.

**I'm having quite a few issues with SurveyJS in Vue:**

- Trying to set the cssType to Bootstrap, throws no errors, following the examples and still no Bootstrap classes or anything remotely related to bootstrap dded

- Trying to set a different button in the dom to be the 'complete' button, throws `Cannot read property of 'style' of null`, again following the examples given

- I'm still curious about why `doOnCurrentPageChnged` is being called twice in the example.

**Comment by Braunson** [comment]
Ok I've removed the Survey logic from the child component and just put it in the Parent component and I have the new 'completed' button working, the bootstrap setting still gets ignored.

I'm looking for a way to re-initialize a survey ... on complete now.

**Comment by andrewtelnov** [comment]
@Braunson Here is [the example with vue + bootstrap](https://surveyjs.io/Examples/Library/?id=bootstrap-theme&platform=Vue)
Here is [the example of restarting the survey](https://plnkr.co/edit/WizdgrkC1R1Sg0CHrXTA?p=preview)

Thank you,
Andrew

**Comment by Braunson** [comment]
Thanks @andrewtelnov I figured another way around it. I'm now working on changing the markup on the radio buttons.

I did notice another odd bug, perhaps not a bug but weird behaviour. How to re-create:

Click YES to Q1
Click YES to Q2 (it should show Q3 a select dropdown)
Click NO to Q1

Then Q2 disappears but Q3 is still shown. I'd expect if the top level parent has changed, all conditional decisions below that should change also, in this case hidden or reset.

I understand I could update my conditional `visibleIf` to take the parent into account but this can get tedious with multiple conditions and multiple levels.

Here's my example https://plnkr.co/edit/K569JMUS0FUs0VC8Hcnm?p=preview

**Comment by andrewtelnov** [comment]
@Braunson Please read the [following docs](https://surveyjs.io/Examples/Library/?id=condition-cascade&platform=Vue#content-docs)

Thank you,
Andrew

**Comment by Braunson** [comment]
Thanks @andrewtelnov I did look through some of examples but I guess the wording threw me off.

**Comment by andrewtelnov** [comment]
@Braunson Yes, we are not a native english speaker here. We are giving our text to a correct time to time.
If you have time to improve this particular document, it would be just great. In this, less people will have the issue that you had.

Thank you,
Andrew

**Comment by Braunson** [comment]
I will help out as much as I can as I learn more of Surveyjs 😄

**Comment by dxcsolarz** [comment]
Braunson,  
I'm having the same issue with bootstrap setting being ignored.   I can't customize the complete button at all.  I'm also using Vue.     Have you had any luck getting this to work?  

Thank you, 
Dave

**Comment by Braunson** [comment]
@dxcsolarz No, I ended up not using SurveyJS.

**Comment by tsv2013** [comment]
@dxcsolarz We've created a live sample - https://plnkr.co/edit/VdjzldPQc7vmkWJTUcCC?p=preview
Does it work for you?

**Comment by dxcsolarz** [comment]
I can get the survey to work, but I want to customize the “complete” button.
I’m using Vue along with Quasar-framework.
Survey-vue version: survey-vue@1.0.30<mailto:survey-vue@1.0.30>

The page looks like this:
[cid:image003.jpg@01D4227A.39F99740]

I’m attempting to change the size, color and shape of the button.

I’m attaching my sample code.    I’ve not had any success to modify css classes to change the complete button.




Dave Solarz
Application Developer
Americas Application & Engineering

Tel: +1 248 639-5397
Cell: +1 734 238-3421

**Comment by dxcsolarz** [comment]
I've tried that and it has no impact on the complete button.

Thanks,

**Comment by dmitrykurmanov** [comment]
@dxcsolarz ok. May be the problem somewhere in your environment. Could you please share a repo with this problem and we will try out to solve it.

**Comment by dmitrykurmanov** [comment]
@dxcsolarz we have https://github.com/surveyjs/surveyjs_vue_quickstart repo. You can fork it and make your changes for faster startup.

**Comment by dxcsolarz** [comment]
Here's the repo: 
[https://github.com/dxcsolarz/TestSurvey](url)

**Comment by dmitrykurmanov** [comment]
@dxcsolarz thanks, I'll check it.

**Comment by dmitrykurmanov** [comment]
@dxcsolarz  I've started your project with `quasar dev` command and can't see the survey. In the console I see the following error: 
![image](https://user-images.githubusercontent.com/22315929/43127654-a64fbcd2-8f38-11e8-98d4-752d8e20aa0f.png)

Could you please check it? May be I did something wrong.

**Comment by dxcsolarz** [comment]
Not sure why you would get this error if your running if from your desktop.
 Did the project start?  
When you execute:   quasar dev
You should see something like: 
` app:dev Running: Mode [ SPA ] with [ MAT ] theme +0ms

 app:quasar-conf Reading quasar.conf.js +2s
 app:dev Checking listening address availability (0.0.0.0:8080)... +0ms
 app:quasar-conf Generating Webpack config +16ms
 app:quasar-conf Extending Webpack config +47ms
 app:generator Generating Webpack entry point +0ms
 app:dev-server Booting up... +15ms

Build completed in 7.602s`

 
http://localhost:8080/surveycomp

**Comment by dmitrykurmanov** [comment]
Thanks for the `/surveycomp` path now I see the survey)

**Comment by dmitrykurmanov** [comment]
@dxcsolarz I've found a several problems:

- you can update your survey-vue to the latest (1.0.34)
- if you want to use bootstrap you need to add `bootstrap.css`
- if you want to use bootstrap you need to remove survey.css import it is for other themes. `// import "survey-vue/survey.css";`
- you set bootstrap theme SurveyVue.StylesManager.applyTheme("bootstrap");  please see the example: 
https://surveyjs.io/Examples/Library?id=questiontype-radiogroup&platform=Vue&theme=bootstrap
- you should remove all code that overrides surveyjs's classes for example:
```
    // vm.survey.onUpdatePanelCssClasses.add(function(survey, options) {
    //   var classes = options.cssClasses;
    //   classes.complete = "btn sv_q_complete_btn";
    // });
```
as a result you get the following situation:

![image](https://user-images.githubusercontent.com/22315929/43152570-884d68da-8f77-11e8-8e3c-6a1afb1f9dcc.png)

So you can override our css usial way.

Please check my fork https://github.com/dmitrykurmanov/TestSurvey and the commit  https://github.com/dmitrykurmanov/TestSurvey/commit/e774d73ff57631783c02d916edeaf3d788cbc2cc

**Comment by dxcsolarz** [comment]
Thanks for looking at this.    It appears the only way to customize the complete button is to use "bootstrap" in the .applyTheme and not use the "survey-vue/survey.css";    My only issue is I lose the original format of the questions when I remove the survey.css.    I will continue to play with it.

Thanks again!

**Comment by dmitrykurmanov** [comment]
@dxcsolarz you are welcome. In fact boostrap.css and survey-vue/survey.css are separate themes. Do you need bootstrap? You can started with our orange theme https://surveyjs.io/Examples/Library?id=questiontype-radiogroup&platform=Vue&theme=orange and then customize it.

**Comment by dxcsolarz** [comment]
Just one last follow up question to your response about using the orange theme and then customize it.   Based on my testing with the link in Plunker you provided, you can't customize the "complete" button size, shape, or even location, correct?

**Comment by tsv2013** [comment]
@dxcsolarz It can easily be done via CSS:
```CSS
.btn.btn-green.btn.sv_complete_btn {
  color: orange;
  background-color: blue;
  border-radius: 20px;
}
.btn.btn-green.btn.sv_complete_btn:hover {
  color: red;
  background-color: yellow;
  border-radius: 0px;
}
```
Here is the working sample - https://plnkr.co/edit/VdjzldPQc7vmkWJTUcCC?p=preview

**Comment by dmitrykurmanov** [comment]
@dxcsolarz 

additional to @tsv2013 's answer:

> Based on my testing with the link in Plunker you provided, you can't customize the "complete" button size, shape, or even location, correct?

You are welcome to do it. But you need to iclude `survey-vue/survey.css` styles instead of `bootstrap.css ` styles and apply "orange" theme:
`SurveyVue.StylesManager.applyTheme("orange");`
